client/src/components/documents/documentView.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import '../../main.scss';
+import NavBar from '../navbar/navBar';
+import Dialog from '../diaLog/confirmDialog';
+import EditDocument from './documentEdit';
+import { currentUser } from '../../helpers/auth';
+import { fetchDocumentById, deleteDocument } from '../../actions/actionCreator';
+
+/**
+ *
+ * @class DocumentView
+ * @extends {Component}
+ */
+class DocumentView extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      loading: true
+    };
+    this.handleDocumentDelete = this.handleDocumentDelete.bind(this);
+  }
+  componentDidMount() {
+    this.handleDocumentView();
+  }
+  handleDocumentView() {
+    const documentId = this.props.params.id;
+    this.props.fetchDocumentById(documentId);
+    this.setState({ loading: false });
+  }
+  /**
+   * @memberOf DocumentView
+   */
+  handleDocumentDelete() {
+    const documentId = this.props.params.id;
+    this.props.deleteDocument(documentId);
+  }
+  render() {
+    const userOwnsDocument = currentUser().id === this.props.document.OwnerId;
+    const modifyButtons = (
```

Comment:
You can take this into a pure function like `renderButton` and call it as `this.renderButton()`. I think that approach is cleaner